src/ngraph/op/pad.cpp

```diff
@@ -174,3 +175,211 @@ std::shared_ptr<Node> op::Pad::get_default_value() const
+    , m_pad_mode{pad_mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v1::Pad::Pad(const Output<Node>& arg,
+                 const Output<Node>& pads_begin,
+                 const Output<Node>& pads_end,
+                 PadMode pad_mode)
+    : Op({arg, pads_begin, pads_end})
+    , m_pad_mode{pad_mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+CoordinateDiff op::v1::Pad::get_pads_begin() const
+{
+    auto pads_begin_node = input_value(1).get_node_shared_ptr();
+    CoordinateDiff pads_begin_coord{};
+    if (auto pads_begin_const = dynamic_pointer_cast<op::Constant>(pads_begin_node))
+    {
+        pads_begin_coord = pads_begin_const->get_vector<ptrdiff_t>();
+    }
+    return pads_begin_coord;
+}
+
+CoordinateDiff op::v1::Pad::get_pads_end() const
+{
+    auto pads_end_node = input_value(2).get_node_shared_ptr();
+    CoordinateDiff pads_end_coord{};
+    if (auto pads_end_const = dynamic_pointer_cast<op::Constant>(pads_end_node))
+    {
+        pads_end_coord = pads_end_const->get_vector<ptrdiff_t>();
+    }
+    return pads_end_coord;
+}
+
+void op::v1::Pad::validate_and_infer_types()
+{
+    element::Type result_et;
+
+    const auto& arg_element_type = get_input_element_type(0);
+    const auto& pads_begin_element_type = get_input_element_type(1);
+    const auto& pads_end_element_type = get_input_element_type(2);
+
+    const auto arg_pad_value_provided = get_input_size() == 4;
+    if (m_pad_mode == PadMode::CONSTANT && arg_pad_value_provided)
+    {
+        const auto& arg_pad_element_type = get_input_element_type(3);
+        const auto& arg_pad_shape = get_input_partial_shape(3);
+        NODE_VALIDATION_CHECK(
+            this,
+            element::Type::merge(result_et, arg_element_type, arg_pad_element_type),
+            "Argument element types do not match (input arg element type: ",
+            arg_element_type,
+            ", arg_pad element type: ",
+            arg_pad_element_type,
+            ").");
+
+        NODE_VALIDATION_CHECK(this,
+                              arg_pad_shape.compatible(PartialShape{}),
+                              "Argument for padding value is not a scalar (shape: ",
+                              arg_pad_shape,
+                              ").");
+    }
+
+    NODE_VALIDATION_CHECK(this,
+                          pads_begin_element_type.compatible(element::Type_t::i64),
+                          "pads_begin must be type i64 (axes type: ",
+                          pads_begin_element_type,
+                          ").");
+
+    NODE_VALIDATION_CHECK(this,
+                          pads_end_element_type.compatible(element::Type_t::i64),
+                          "pads_end must be type i64 (axes type: ",
+                          pads_end_element_type,
+                          ").");
+
+    const auto& pads_begin_shape = get_input_partial_shape(1);
+    const auto& pads_begin_rank = pads_begin_shape.rank();
+    if (pads_begin_rank.is_static())
```

Comment:
You can collapse this and the check inside the `if` down to `pads_begin_rank.compatible(1)`, but no matter.